src/modules/Game.class.js

```diff
@@ -50,19 +138,139 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    if (!this.isGameStarted) {
+      return 'idle';
+    }
+
+    const hasWinningCell = this.board.some((row) => {
+      return row.some((cell) => cell === 2048);
+    });
+
+    if (hasWinningCell) {
+      return 'win'; // Игрок выиграл
+    }
+
+    const hasMoves = this.checkAvailableMoves();
+
+    if (!hasMoves) {
+      return 'lose';
+    }
+
+    return 'playing';
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    const randomCell = this.getRandomCell(2);
+
+    this.addNewCell(randomCell);
+    this.isGameStarted = true;
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.board = this.initialState.map((row) => [...row]);
+    this.score = 0;
+    this.isGameStarted = false;
+  }
 
   // Add your own methods here
+  getRandomCell(count) {
+    const currentBoard = this.getState();
+
+    const emptyCells = [];
+
+    for (let row = 0; row < currentBoard.length; row++) {
+      for (let col = 0; col < currentBoard[row].length; col++) {
+        if (currentBoard[row][col] === 0) {
+          emptyCells.push({ row, col });
+        }
+      }
+    }
+
+    const cellsToAdd = Math.min(count, emptyCells.length);
+
+    const randomCells = [];
+
+    while (randomCells.length < cellsToAdd) {
+      const randomIndex = Math.floor(Math.random() * emptyCells.length);
+
+      randomCells.push(emptyCells[randomIndex]);
+      emptyCells.splice(randomIndex, 1);
+    }
+
+    return randomCells;
+  }
+
+  addNewCell(newIndex) {
+    for (const cell of newIndex) {
+      const row = cell.row;
+      const col = cell.col;
+
+      this.board[row][col] = Math.random() < 0.9 ? 2 : 4;
+    }
+  }
+
+  checkAvailableMoves() {
+    const hasOs = this.board.some((row) => row.some((cell) => cell === 0));
+
+    const movesHorisontsl = this.board.some((row) => {
+      return row.some((cell, index) => cell === row[index + 1]);
+    });
+    const transposedBoard = this.transpose(this.board);
+    const movesVertical = transposedBoard.some((row) => {
+      return row.some((cell, index) => cell === row[index + 1]);
+    });
+
+    return hasOs || movesHorisontsl || movesVertical;
+  }
+
+  joinValues(arr, targetLength) {
+    const result = [];
+    let scoreIncrease = 0;
+
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] === arr[i + 1]) {
+        const mergedValue = arr[i] * 2;
+
+        result.push(mergedValue); // Удваиваем значение
```

Comment:
Remove comments everywhere 
```suggestion
        result.push(mergedValue);
```